### ISSUE TITLE
Got a "panic: runtime error: invalid memory address or nil pointer dereference" when a new DC joins the Mesh

### ISSUE THREAD

**Issue by mariusehr1** [hashicorp/consul#8753]
#### Overview of the Issue
Hello,

I have a Mesh with three datacenters, they are all working together without problem.
I am facing an issue when my new kubernetes datacenter tries to join the Mesh (deployed with the helm chart) : I get the following error : 

```
Sep 25 12:53:15 ip-172-16-32-10 consul[4487]: panic: runtime error: invalid memory address or nil pointer dereference
Sep 25 12:53:15 ip-172-16-32-10 consul[4487]: [signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x1a8648a]
Sep 25 12:53:15 ip-172-16-32-10 consul[4487]: goroutine 302 [running]:
Sep 25 12:53:15 ip-172-16-32-10 consul[4487]: github.com/envoyproxy/go-control-plane/envoy/api/v2.(*Cluster).MarshalTo(0x0, 0x54519f8, 0x0, 0x0, 0x29b9f60, 0x1, 0x54519f8)
Sep 25 12:53:15 ip-172-16-32-10 consul[4487]:         /go/pkg/mod/github.com/envoyproxy/go-control-plane@v0.8.0/envoy/api/v2/cds.pb.go:2912 +0x3a
Sep 25 12:53:15 ip-172-16-32-10 consul[4487]: github.com/envoyproxy/go-control-plane/envoy/api/v2.(*Cluster).XXX_Marshal(0x0, 0x54519f8, 0x0, 0x0, 0x0, 0xc0006fd501, 0x49, 0x49, 0x0, 0x0)
Sep 25 12:53:15 ip-172-16-32-10 consul[4487]:         /go/pkg/mod/github.com/envoyproxy/go-control-plane@v0.8.0/envoy/api/v2/cds.pb.go:490 +0x48
Sep 25 12:53:15 ip-172-16-32-10 consul[4487]: github.com/gogo/protobuf/proto.Marshal(0x384bb80, 0x0, 0xc0006fd590, 0x49, 0x49, 0x0, 0x0)
Sep 25 12:53:15 ip-172-16-32-10 consul[4487]:         /go/pkg/mod/github.com/gogo/protobuf@v1.2.2-0.20190723190241-65acae22fc9d/proto/table_marshal.go:2939 +0x2d6
Sep 25 12:53:15 ip-172-16-32-10 consul[4487]: github.com/hashicorp/consul/agent/xds.createResponse(0x32a5ff7, 0x28, 0xc0000f8d70, 0x8, 0xc0000f8d60, 0x8, 0xc000e75080, 0xa, 0xc, 0x454d4b, ...)
Sep 25 12:53:15 ip-172-16-32-10 consul[4487]:         /home/circleci/project/consul/agent/xds/response.go:21 +0x3a9
Sep 25 12:53:15 ip-172-16-32-10 consul[4487]: github.com/hashicorp/consul/agent/xds.(*xDSType).SendIfNew(0xc0004780f0, 0xc00011f180, 0x20, 0xc0008fd930, 0xc000c4db40, 0x0)
Sep 25 12:53:15 ip-172-16-32-10 consul[4487]:         /home/circleci/project/consul/agent/xds/server.go:433 +0x26a
Sep 25 12:53:15 ip-172-16-32-10 consul[4487]: github.com/hashicorp/consul/agent/xds.(*Server).process(0xc00075d2c0, 0x388dc00, 0xc0003f9f80, 0xc0004e13e0, 0x0, 0x0)
Sep 25 12:53:15 ip-172-16-32-10 consul[4487]:         /home/circleci/project/consul/agent/xds/server.go:369 +0x832
Sep 25 12:53:15 ip-172-16-32-10 consul[4487]: github.com/hashicorp/consul/agent/xds.(*Server).StreamAggregatedResources(0xc00075d2c0, 0x388dc00, 0xc0003f9f80, 0x382c680, 0xc00075d2c0)
Sep 25 12:53:15 ip-172-16-32-10 consul[4487]:         /home/circleci/project/consul/agent/xds/server.go:171 +0xc5
Sep 25 12:53:15 ip-172-16-32-10 consul[4487]: github.com/envoyproxy/go-control-plane/envoy/service/discovery/v2._AggregatedDiscoveryService_StreamAggregatedResources_Handler(0x319c3e0, 0xc00075d2c0, 0x3883a60, 0xc00053c300, 0x54519f8, 0xc00010fb00)
Sep 25 12:53:15 ip-172-16-32-10 consul[4487]:         /go/pkg/mod/github.com/envoyproxy/go-control-plane@v0.8.0/envoy/service/discovery/v2/ads.pb.go:197 +0xad
Sep 25 12:53:15 ip-172-16-32-10 consul[4487]: google.golang.org/grpc.(*Server).processStreamingRPC(0xc0000ca840, 0x3893e40, 0xc0004f8780, 0xc00010fb00, 0xc000132000, 0x53f0360, 0x0, 0x0, 0x0)
Sep 25 12:53:15 ip-172-16-32-10 consul[4487]:         /go/pkg/mod/google.golang.org/grpc@v1.23.0/server.go:1199 +0xd21
Sep 25 12:53:15 ip-172-16-32-10 consul[4487]: google.golang.org/grpc.(*Server).handleStream(0xc0000ca840, 0x3893e40, 0xc0004f8780, 0xc00010fb00, 0x0)
Sep 25 12:53:15 ip-172-16-32-10 consul[4487]:         /go/pkg/mod/google.golang.org/grpc@v1.23.0/server.go:1279 +0xcd6
Sep 25 12:53:15 ip-172-16-32-10 consul[4487]: google.golang.org/grpc.(*Server).serveStreams.func1.1(0xc000517be0, 0xc0000ca840, 0x3893e40, 0xc0004f8780, 0xc00010fb00)
Sep 25 12:53:15 ip-172-16-32-10 consul[4487]:         /go/pkg/mod/google.golang.org/grpc@v1.23.0/server.go:710 +0xa1
Sep 25 12:53:15 ip-172-16-32-10 consul[4487]: created by google.golang.org/grpc.(*Server).serveStreams.func1
Sep 25 12:53:15 ip-172-16-32-10 consul[4487]:         /go/pkg/mod/google.golang.org/grpc@v1.23.0/server.go:708 +0xa1
Sep 25 12:53:15 ip-172-16-32-10 systemd[1]: consul-agent-server-dc1.service: Main process exited, code=exited, status=2/INVALIDARGUMENT
Sep 25 12:53:15 ip-172-16-32-10 systemd[1]: consul-agent-server-dc1.service: Failed with result 'exit-code'.
```
This will occur on all of my server agent and thus make my whole mesh go down. It is very random, sometimes a restart of the systemd unit will suffice, sometimes I will have to reprovision the machines.

#### Reproduction Steps

Steps to reproduce this issue, eg:

1. Create a cluster with 3 datacenters, each of them has a master and 1 to 3 services
1. Deploy the Helm chart on a new EKS or any Kubernetes instance : the mesh gateways within my kubernetes will fail on the first time, I configure this datacenter as secondary and I give him the primary gateway IP as well.
1. View error

### Consul info for both my datacenters

<details>
  <summary>Primary DC info (he is connected with two other DCs)</summary>

```
agent:
        check_monitors = 0
        check_ttls = 0
        checks = 2
        services = 2
build:
        prerelease = 
        revision = 3111cb8c
        version = 1.8.0
consul:
        acl = disabled
        bootstrap = true
        known_datacenters = 3
        leader = true
        leader_addr = 172.16.32.10:8300
        server = true
raft:
        applied_index = 142
        commit_index = 142
        fsm_pending = 0
        last_contact = 0
        last_log_index = 142
        last_log_term = 2
        last_snapshot_index = 0
        last_snapshot_term = 0
        latest_configuration = [{Suffrage:Voter ID:dc8df05e-fae1-a907-b957-3a091bbbbf82 Address:172.16.32.10:8300}]
        latest_configuration_index = 0
        num_peers = 0
        protocol_version = 3
        protocol_version_max = 3
        protocol_version_min = 0
        snapshot_version_max = 1
        snapshot_version_min = 0
        state = Leader
        term = 2
runtime:
        arch = amd64
        cpu_count = 1
        goroutines = 270
        max_procs = 1
        os = linux
        version = go1.14.4
serf_lan:
        coordinate_resets = 0
        encrypted = true
        event_queue = 0
        event_time = 2
        failed = 0
        health_score = 0
        intent_queue = 0
        left = 0
        member_time = 3
        members = 3
        query_queue = 0
        query_time = 1
serf_wan:
        coordinate_resets = 0
        encrypted = true
        event_queue = 0
        event_time = 1
        failed = 1
        health_score = 0
        intent_queue = 0
        left = 0
        member_time = 3
        members = 3
        query_queue = 0
        query_time = 1
```

</details>

<details>
  <summary>Kubernetes DC info</summary>

```
agent:
        check_monitors = 0
        check_ttls = 0
        checks = 0
        services = 0
build:
        prerelease = 
        revision = 3111cb8c
        version = 1.8.0
consul:
        acl = disabled
        bootstrap = false
        known_datacenters = 4
        leader = true
        leader_addr = 192.168.49.48:8300
        server = true
raft:
        applied_index = 575
        commit_index = 575
        fsm_pending = 0
        last_contact = 0
        last_log_index = 575
        last_log_term = 4
        last_snapshot_index = 0
        last_snapshot_term = 0
        latest_configuration = [{Suffrage:Voter ID:27401414-0ed4-243a-e7b4-ed1582b163ab Address:192.168.42.192:8300} {Suffrage:Voter ID:412767d1-522b-22c4-74cc-5c0dc43d811f Address:192.168.10.222:8300}]
        latest_configuration_index = 0
        num_peers = 1
        protocol_version = 3
        protocol_version_max = 3
        protocol_version_min = 0
        snapshot_version_max = 1
        snapshot_version_min = 0
        state = Leader
        term = 4
runtime:
        arch = amd64
        cpu_count = 2
        goroutines = 134
        max_procs = 2
        os = linux
        version = go1.14.4
serf_lan:
        coordinate_resets = 0
        encrypted = true
        event_queue = 0
        event_time = 2
        failed = 0
        health_score = 0
        intent_queue = 0
        left = 0
        member_time = 14
        members = 4
        query_queue = 0
        query_time = 1
serf_wan:
        coordinate_resets = 0
        encrypted = true
        event_queue = 0
        event_time = 1
        failed = 1
        health_score = 0
        intent_queue = 0
        left = 0
        member_time = 12
        members = 5
        query_queue = 0
        query_time = 1
```

</details>

### Operating system and Environment details

Running Ubuntu 20.04 with systemd units on my 3 first DC

The Kubernetes cluster is running the default EKS AMI.

The consul version is 1.8.0 on both environments

Let me know if you need more informations.

Thanks.

Marius

**Comment by mkeeler** [comment]
@mariusehr1 Does the same thing happen with Consul 1.8.4? I know there were some similarly looking issues either in 1.8.0 or 1.8.1 that have since been fixed.

**Comment by dnephin** [comment]
This does look like the same stack trace as #8155, which was fixed in 1.8.1

**Comment by mariusehr1** [comment]
Thanks  for your very quick answers.
It looks fine now, i'll remember to use the last version.
